validation/src/ma5_validation/utils/job_handler.py

```diff
@@ -0,0 +1,219 @@
+                    + self.script.cpp
+                    + " "
+                    + os.path.join(
+                        self.log_path, self.script.name, "Build/SampleAnalyzer/User/Analyzer"
+                    )
+                    + "/."
+                )
+                print("   * Running: " + command)
+                os.system(command)
+
+                command = (
+                    "cp "
+                    + self.script.header
+                    + " "
+                    + os.path.join(
+                        self.log_path, self.script.name, "Build/SampleAnalyzer/User/Analyzer"
+                    )
+                    + "/."
+                )
+                print("   * Running: " + command)
+                os.system(command)
+
+                # Execute
+                os.system(
+                    "source setup.sh && make clean all &> compilation.log && "
+                    + "./MadAnalysis5job "
+                    + os.path.join(self.log_path, self.script.name, "Input/_defaultset.list")
+                    + self.script.command_line
+                    + " &> "
+                    + self.log_file
+                )
+            except Exception as err:
+                log_file = ""
+                with open(self.log_file, "r", encoding="utf-8") as log:
+                    log_file = log.read()
+                raise MadAnalysis5ExecutionError(
+                    f"A problem has occured during MadAnalysis 5 execution\n\n{err}\n\n{log_file}"
+                )
+
+        return True
+
+    def check(self):
+        """
+        Check the log file for errors
+
+        Raises
+        ------
+        MadAnalysis5Error
+            If MadAnalysis 5 raised an error during the execution.
+        """
+        beginTag = False
+        errorTag = False
+        log_file = None
+        with open(self.log_file, "r", encoding="utf-8") as log:
+            for line in log:
+                # Start from the beginning of the validation script to validate the session instead of the end.
+                # This will give the ability to check the errors on the python-interface as well
+                if line.find("ma5>#BEGIN") != -1:
+                    beginTag = True
+                if not self.script.IsExpert:
+                    if beginTag:
+                        if line.find("ERROR") != -1 or line.find("MA5-ERROR") != -1:
+                            errorTag = True
+                            log_file = log.read()
+                            break
+        if not beginTag and not self.script.IsExpert:
+            InvalidSyntax("   * Can not find the beginning of the script.")
+
+        if errorTag:
+            raise MadAnalysis5Error(
+                f"MadAnalysis has raised an error. For details, please see: {self.log_file}"
+                f"\n\n\n{log_file}"
+            )
+
+        # TODO: This checker needs an numerical validation as well. Currently it only checks
```

Comment:
@econte-cms This is the location where we can implement numerical validations. `script_handler.py` can additionally take the `#COMPARE ...` keyword where the files to be compared can be located. This way everything can be robust.

Reply:
This is interesting. Do we have a working example?